crates/dbs-address-space/src/memory/hybrid.rs

```diff
@@ -363,3 +363,719 @@ impl<B: Bitmap + 'static> GuestMemory for GuestMemoryHybrid<B> {
+        ));
+
+        // Error invalid backend address case in laod() on raw region.
+        assert!(matches!(
+            guest_raw_region
+                .load::<u64>(invalid_addr, Ordering::Relaxed)
+                .err()
+                .unwrap(),
+            GuestMemoryError::InvalidBackendAddress
+        ));
+    }
+
+    #[test]
+    fn test_bitmap() {
+        // TODO: Need futher and detailed test on bitmap object.
+        let start_addr = GuestAddress(0x0);
+        let mmap_reg =
+            GuestRegionMmap::new(MmapRegion::<()>::new(0x800).unwrap(), start_addr).unwrap();
+        let guest_mmap_region = GuestRegionHybrid::from_mmap_region(mmap_reg);
+        let mut buf_of_raw_region = [0u8; 0x800];
+        let raw_region = unsafe {
+            GuestRegionRaw::<()>::new(start_addr, &mut buf_of_raw_region as *mut _, 0x800)
+        };
+        let guest_raw_region = GuestRegionHybrid::from_raw_region(raw_region);
+
+        assert_eq!(guest_mmap_region.bitmap(), guest_raw_region.bitmap());
+    }
+
+    #[test]
+    fn test_get_host_address_on_mmap_region() {
+        let start_addr = GuestAddress(0x0);
+        let mmap_reg =
+            GuestRegionMmap::new(MmapRegion::<()>::new(0x800).unwrap(), start_addr).unwrap();
+        let guest_region = GuestRegionHybrid::from_mmap_region(mmap_reg);
+
+        // Normal case.
+        let addr_1 = guest_region
+            .get_host_address(MemoryRegionAddress(0x0))
+            .unwrap();
+        let addr_2 = guest_region
+            .get_host_address(MemoryRegionAddress(0x400))
+            .unwrap();
+        assert_eq!(addr_1 as u64 + 0x400, addr_2 as u64);
+
+        // Error invalid backend address case.
+        let invalid_addr = MemoryRegionAddress(0x900);
+        assert!(matches!(
+            guest_region.get_host_address(invalid_addr).err().unwrap(),
+            GuestMemoryError::InvalidBackendAddress
+        ));
+    }
+
+    #[test]
+    fn test_get_host_address_on_raw_region() {
+        let start_addr = GuestAddress(0x0);
+        let mut buf_of_raw_region = [0u8; 0x800];
+        let raw_region = unsafe {
+            GuestRegionRaw::<()>::new(start_addr, &mut buf_of_raw_region as *mut _, 0x800)
+        };
+        let guest_region = GuestRegionHybrid::from_raw_region(raw_region);
+
+        // Normal case.
+        let addr_1 = guest_region
+            .get_host_address(MemoryRegionAddress(0x0))
+            .unwrap();
+        let addr_2 = guest_region
+            .get_host_address(MemoryRegionAddress(0x400))
+            .unwrap();
+        assert_eq!(addr_1 as u64 + 0x400, addr_2 as u64);
+
+        // Error invalid backend address case.
+        let invalid_addr = MemoryRegionAddress(0x900);
+        assert!(matches!(
+            guest_region.get_host_address(invalid_addr).err().unwrap(),
+            GuestMemoryError::InvalidBackendAddress
+        ));
+    }
+
+    // TODO: The following function are not yet implemented:
```

Comment:
If there is a TODO in the code, please create an issue for this and add issue number here.